src/analysis/cfg-impl.h

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2023 WebAssembly Community Group participants
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef wasm_analysis_cfg_impl_h
+#define wasm_analysis_cfg_impl_h
+
+#include "cfg.h"
+
+namespace wasm::analysis {
+
+struct BasicBlock::Predecessors {
+  struct iterator;
+  iterator begin() const;
+  iterator end() const;
+
+  bool operator==(const Predecessors& other) const {
+    return cfg == other.cfg && index == other.index;
+  }
+  bool operator!=(const Predecessors& other) const { return !(*this == other); }
+
+private:
+  const CFG* cfg;
```

Comment:
It's more convenient to use pointers so I can use list initialization rather than having to declare a constructor. For some reason list initialization does not work with references.

Reply:
This code no longer exists.